### DISCUSSION TITLE
Suggested Communication Source to use for streaming >15 seconds Audio

### DISCUSSION THREAD

**Discussion by soham2560** [pschatzmann/arduino-audio-tools#1586] [Q&A] [answered]
I am using an ESP32 LOLIN D32 for a certain project which does the following.
- Audio is recorded from an I2S microphone and streamed to a server
- The ESP32 gets a short custom audio reply generated on the server

I have achieved the first part using websockets, but am not able to use the same for the second part. On output the audio cuts a lot and comes in chunks. I am aiming to move to some other pre-existing source from [here](https://github.com/pschatzmann/arduino-audio-tools/wiki/Communication). Which protocol is suggested to be used if

- Less than 15 Seconds of converted Audio at a time
- 16bit mono audio is good enough
- Latency is not that big of a priority, but no data should be lost, i.e continuous audio without hiccups
- If decoders should be used do suggest which one might be useful
- ESP32 might be on networks with low speed (>1MBPS), and also maybe on mobile hotspots at some times

This implementation is to be expanded to multiple devices later, and the audio stream can be open to public, would prefer that it is selectively accessible

**Comment by pschatzmann** [comment, marked as answer]
It is my gut feeling that the whole scenario could be handled by a simple http post from the ESP32 to the server which would provide the audio reply in the same request. 

Keep the sample rate as low as possible.

To start I would suggest to try out the following
- start with a sample rate of 8000 (this shoud be usually good enough)
- encoding
  - use no encoding 16 bits
  - you can half the data by using no encoding 8 bits 
  - try out e.g. the adpcm codec

**Reply by soham2560** [reply]
that actually sounds efficient, I got the GET request working but I will give this a try too
